airflow/providers/apache/spark/pyspark_submit_script.jinja2

```diff
@@ -0,0 +1,60 @@
+{#
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+-#}
+from __future__ import annotations
+
+import {{ pickling_library }}
+import sys
+from pyspark import SparkFiles
+from pyspark.sql import SparkSession
+
+{% if expect_airflow %}
+ {# Check whether Airflow is available in the environment.
+ # If it is, we'll want to ensure that we integrate any macros that are being provided
+ # by plugins prior to unpickling the task context. #}
+if sys.version_info >= (3,6):
+    try:
+        from airflow.plugins_manager import integrate_macros_plugins
+        integrate_macros_plugins()
+    except ImportError:
+        {# Airflow is not available in this environment, therefore we won't
+         # be able to integrate any plugin macros. #}
+        pass
+{% endif %}
+
+# Script
+{{ python_callable_source }}
+
+{% if use_arguments %}
+SparkSession.builder.getOrCreate()
+with open(SparkFiles.get("{{ input_filename }}"), "rb") as file:
+    arg_dict = {{ pickling_library }}.load(file)
+{% else %}
+arg_dict = {"args": [], "kwargs": {}}
+{% endif %}
+
+{% if use_spark_session %}
+arg_dict["kwargs"]["spark_session"] = SparkSession.builder.getOrCreate()
+{% endif %}
+
+{% if use_spark_context %}
+arg_dict["kwargs"]["spark_context"] = SparkSession.builder.getOrCreate().sparkContext
+{% endif %}
```

Comment:
I have modified the part of the code that parses strings to use Jinja2 templates instead. 
I will update and refactor the tests accordingly